include/envoy/api/io_error.h

```diff
@@ -25,6 +25,8 @@ class IoError {
     InProgress,
     // Permission denied.
     Permission,
+    // Bad handle
```

Comment:
This was covered in a different @danzh2010 PR, but can we remove this? This should never happen and should be covered with ASSERT/RELEASE_ASSERT if necessary depending on the situation..

Reply:
Done.